libs/ngfire/firestore/src/collection.ts

```diff
@@ -46,16 +46,22 @@ export abstract class FireCollection<E extends DocumentData> {
   protected useCache<T extends E>(ref: DocumentReference<T>): Observable<T>
   protected useCache<T extends E>(ref: Query<T>): Observable<T[]>
   protected useCache<T extends E>(ref: DocumentReference<T> | Query<T>): Observable<T | T[]>   
-  protected useCache<T extends E>(ref: DocumentReference<T> | Query<T>): Observable<T | T[]> {    
-    if (!this.memorize) return fromRef(ref as Query<T>).pipe(map(snap => this.snapToData(snap)));
-    const transfer = this.firestore.getTransfer(ref);
-    const initial = this.firestore.getState(ref);
-    const snap$ = this.firestore.fromMemory(ref, this.delayToUnsubscribe).pipe(
-      tap(snap => this.firestore.setState(ref, snap))
-    );
-    if (transfer) return snap$.pipe(map(snap => this.snapToData(snap)), startWith(transfer));
-    if (initial) return snap$.pipe(startWith(initial), map(snap => this.snapToData(snap)));
-    return snap$.pipe(map(snap => this.snapToData(snap)));
+  protected useCache<T extends E>(ref: DocumentReference<T> | Query<T>): Observable<T | T[]> {   
+      if (!this.memorize) {
+        return this.zone.runOutsideAngular(() => fromRef(ref as Query<T>)).pipe(
+          map(snap => this.snapToData(snap)),
+          keepUnstableUntilFirst(this.zone)
+        );
+      }
+      const transfer = this.firestore.getTransfer(ref);
+      const initial = this.firestore.getState(ref);
+      const snap$ = this.zone.runOutsideAngular(() => this.firestore.fromMemory(ref, this.delayToUnsubscribe)).pipe(
+        tap(snap => this.firestore.setState(ref, snap)),
+        keepUnstableUntilFirst(this.zone)
+      );
+      if (transfer) return snap$.pipe(map(snap => this.snapToData(snap)), startWith(transfer));
+      if (initial) return snap$.pipe(startWith(initial), map(snap => this.snapToData(snap)));
+      return snap$.pipe(map(snap => this.snapToData(snap)));
```

Comment:
I would rather put it in the end
```suggestion
      if (!this.memorize) {
        return this.zone.runOutsideAngular(() => fromRef(ref as Query<T>)).pipe(
          map(snap => this.snapToData(snap)),
          keepUnstableUntilFirst(this.zone)
        );
      }
      const transfer = this.firestore.getTransfer(ref);
      const initial = this.firestore.getState(ref);
      const snap$ = this.firestore.fromMemory(ref, this.delayToUnsubscribe).pipe(
        tap(snap => this.firestore.setState(ref, snap))
      );
      if (transfer) return snap$.pipe(map(snap => this.snapToData(snap)), startWith(transfer));
      if (initial) return snap$.pipe(startWith(initial), map(snap => this.snapToData(snap)));
      return this.zone.runOutsideAngular(() => snap$.pipe(
        map(snap => this.snapToData(snap))
        keepUnstableUntilFirst(this.zone)
       ));
```

Reply:
Then it wouldn't cover the case where `transfer` or `initial` is defined